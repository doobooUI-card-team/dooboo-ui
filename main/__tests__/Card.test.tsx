import * as React from 'react';

import { Text, View } from 'react-native';
import { Card } from '..';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

// eslint-disable-next-line
const component = (props?: any): React.ReactElement => {
  return <Card {...props} />;
};

describe('[Card]', () => {
  let rendered: renderer.ReactTestRenderer;
  let root: renderer.ReactTestInstance;

  it('should render without crashing', () => {
    rendered = renderer.create(component());
    expect(rendered.toJSON()).toMatchSnapshot();
    expect(rendered.toJSON()).toBeTruthy();
  });

  it('should render loading status', () => {
    rendered = renderer.create(component({ loading: true }));
    root = rendered.root;

    const texts = root.findAllByType(Text);
    expect(texts).toHaveLength(0);
  });

  it('should render title and subtitle', () => {
    rendered = renderer.create(
      component({ title: 'Card title', subTitle: 'Card subTitle' }),
    );
    root = rendered.root;

    const texts = root.findAllByType(Text);
    expect(texts).toHaveLength(2);
  });

  it('should render children', () => {
    rendered = renderer.create(
      component({
        children: (
          <View>
            <Text>first children</Text>
            <Text>second children</Text>
          </View>
        ),
      }),
    );
    root = rendered.root;

    const texts = root.findAllByType(Text);
    expect(texts).toHaveLength(2);
  });

  it('should simulate props', (): void => {
    const rendered = renderer.create(
      component({
        testID: 'CARD_ID',
        title: 'This is title',
        subTitle: 'This is subTitle',
        children: <Text>This is children</Text>,
      }),
    );

    rendered.update(component({ hasDivider: false }));
    expect(rendered).toMatchSnapshot();
    expect(rendered).toBeTruthy();

    rendered.update(component({ raised: true }));
    expect(rendered).toMatchSnapshot();
    expect(rendered).toBeTruthy();

    rendered.update(component({ outlined: true }));
    expect(rendered).toMatchSnapshot();
    expect(rendered).toBeTruthy();
  });
});
