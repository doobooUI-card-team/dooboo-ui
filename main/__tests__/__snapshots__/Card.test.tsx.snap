// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Card] should render without crashing 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "flex-start",
        "backgroundColor": "#fff",
        "elevation": 5,
        "flexDirection": "column",
        "shadowColor": "#000",
        "shadowOpacity": 0.25,
        "shadowRadius": 4,
        "width": 256,
      },
    ]
  }
/>
`;

exports[`[Card] should simulate props 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "flex-start",
        "backgroundColor": "#fff",
        "elevation": 5,
        "flexDirection": "column",
        "shadowColor": "#000",
        "shadowOpacity": 0.25,
        "shadowRadius": 4,
        "width": 256,
      },
    ]
  }
/>
`;

exports[`[Card] should simulate props 2`] = `
<View
  raised={true}
  style={
    Array [
      Object {
        "alignItems": "flex-start",
        "backgroundColor": "#fff",
        "elevation": 10,
        "flexDirection": "column",
        "shadowColor": "#000",
        "shadowOpacity": 0.34,
        "shadowRadius": 6.27,
        "width": 256,
      },
    ]
  }
/>
`;

exports[`[Card] should simulate props 3`] = `
<View
  outlined={true}
  style={
    Array [
      Object {
        "alignItems": "flex-start",
        "backgroundColor": "#fff",
        "borderColor": "#ccc",
        "borderWidth": 1,
        "flexDirection": "column",
        "width": 256,
      },
    ]
  }
/>
`;
